build_simulators.sh

```diff
@@ -1,14 +1,8 @@
 #!/bin/bash
 
-function test_build () {
-    declare -i rc=0
-    eval $*
-    rc=$?
-    if [ $rc -ne 0 ]; then
-        echo "Failure to execute: $*"
-        exit $rc
-    fi
-}
+set -e
 
-test_build make ARCH=RV32 c_emulator/riscv_sim_RV32
-test_build make ARCH=RV64 c_emulator/riscv_sim_RV64
+mkdir -p build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Release
+make -j2 csim
```

Comment:
why j2? Maybe something like `-j$(nproc || echo 2)`?

Reply:
nproc isn't a thing on macOS (unless you install GNU coreutils)

(it does exist on FreeBSD these days though)

Reply:
yeah that's why I added the fallback. I guess you could do something like this:
```suggestion
make -j$(nproc || sysctl -n hw.ncpu || echo 2) csim
```

Reply:
That verbatim will shout at you over stderr though...

Reply:
Yeah will need the right redirects. But then again I'm not sure this script really matters

Reply:
I picked 2 because it builds 2 simulators, and most of the time is just compiling the Sail, the `model.c` and linking which are all serial for each simulator.

I agree this script doesn't really matter though - it's probably more important as a kind of "how to compile" than something I'd expect people to actually use, in which case keeping it simple is probably more important than optimal CPU use? Happy to change it to e.g. 4 though.

Reply:
Probably doesn't matter. Keeping two for the two simulators is fine IMO.